### PR TITLE
615 dns migration

### PR DESCRIPTION
**Pull request by smneal**
## Context

We need to migrate the domain from Route 53 to Azure DNS so we can manage the services domains using Terraform.

## Changes proposed in this pull request

- Add Makefile commands to enable deployment of resource group, state file storage, front door and the DNS zone.
- Add Terraform code to support consuming the infrastructure and environment domains modules
- Add global_config files

## Guidance for review

Ensure resource group and resources are deployed as required. These are available [here](https://portal.azure.com/#@platform.education.gov.uk/resource/subscriptions/3c033a0c-7a1c-4653-93cb-0f2a9f57a391/resourceGroups/s189p01-cpdinf-dom-rg/overview)

## Next steps

- [ ] Get the _dnsauth record added to Route 53 so the apex domain certificate is validated.
- [ ] Test the domain following https://vninja.net/2020/02/06/macos-custom-dns-resolvers
- [ ] Agree migration date with service team
- [ ] Raise service request and then change to update the R53 CNAME record to an NS record

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	az deployment sub create --name "resourcedeploy-tsc-$(shell date +%Y%m%d%H%M%S)" \
+		-l "${REGION}" --template-uri "https://raw.githubusercontent.com/DFE-Digital/tra-shared-services/${ARM_TEMPLATE_TAG}/azure/resourcedeploy.json" \
+		--parameters "resourceGroupName=${RESOURCE_GROUP_NAME}" 'tags=${RG_TAGS}' \
+		"tfStorageAccountName=${STORAGE_ACCOUNT_NAME}" "tfStorageContainerName=terraform-state" \
+		${KV_ARG} \
+		${KV_DIAG_ARG} \
+		"enableKVPurgeProtection=${KV_PURGE_PROTECTION}" \
+		${WHAT_IF}
+
+deploy-arm-resources: arm-deployment
```
Comment (nickdevdfe) [inline Makefile:78]: This appears to be redundant, ie line 65

File: terraform/domains/environment_domains/config/production.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "hosted_zone": {
+    "professional-development-for-teachers-leaders.education.gov.uk": {
+      "front_door_name": "s189p01-cpdinf-dom-fd",
+      "resource_group_name": "s189p01-cpdinf-dom-rg",
```
Comment (nickdevdfe) [inline terraform/domains/environment_domains/config/production.tfvars.json:5]: Isn't the resource group name supposed to be the format s189p01-<shortname>domains-rg?

Comment (smneal) [reply terraform/domains/environment_domains/config/production.tfvars.json:5]: Not now. Going forward they will change to this more standard format. We can reuse more code if we standardised the naming.

File: Makefile (hunk 1 above)
Comment (smneal) [reply Makefile:78]: This is from the new service template and may get added to later, so we'll leave it in for now to keep arm-deployment as re-usable as possible.

**Comment by smneal** [comment]
[Create DNS zone in Azure DNS via terraform and migrate](https://trello.com/c/5c9mEnnI/615-create-dns-zone-in-azure-dns-via-terraform-and-migrate)

File: global_config/development.sh
Comment (saliceti) [inline global_config/development.sh:1]: As discussed, there is no need for a developoment custom domain

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,119 @@
+TERRAFILE_VERSION=0.8
+ARM_TEMPLATE_TAG=1.1.10
+RG_TAGS={"Product" : "Teacher services cloud"}
```
Comment (saliceti) [inline Makefile:3]: Do we need a specific product?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+REGION=UK South
+SERVICE_NAME=cpd-information
+SERVICE_SHORT=cpdinf
+DOCKER_REPOSITORY=#DOCKER_REPOSITORY#
```
Comment (saliceti) [inline Makefile:7]: Remove

**Review by saliceti — APPROVED** [review]

**Pull request merged**